analyzers/src/SonarAnalyzer.Common/Helpers/WildcardPatternMatcher.cs

```diff
@@ -25,70 +25,54 @@
 
 namespace SonarAnalyzer.Helpers;
 
-internal static class WildcardPatternMatcher
+internal class WildcardPatternMatcher
 {
-    public static bool IsMatch(string pattern, string input) =>
+    public static bool IsMatch(string pattern, string input, bool timoutFallback = false) =>
```

Comment:
I'd add `Result` to the name, as `timeoutFallback` isn't clear enough. Even when I usually prefer shorter names.
I'd not give it a default value. It is used twice, each time with different value.
```suggestion
    public static bool IsMatch(string pattern, string input, bool timoutFallbackResult) =>
```